azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -71,11 +79,21 @@ def trust_add(self, instance_name: str, resource_group: str, file: str, secret_n
         secret_name = secret_name if secret_name else file_name.replace(".", "-")
 
         # iterate over secrets to check if secret with same name exists
-        secret_name = self._check_secret_name(secrets, secret_name, spc_keyvault_name, "secret")
+        secret_name = self._check_secret_name(
+            secret_names=[secret.id for secret in secrets],
+            secret_name=secret_name,
+            spc_keyvault_name=spc_keyvault_name,
+            flag="secret-name",
+            overwrite_secret=overwrite_secret,
+        )
+
+        if not secret_name:
```

Comment:
What does the user do if this condition is hit ?

Reply:
the above `_check_secret_name` will return None if when user chose to skip overwriting the secret(if secret with same name detected in the key vault), once they skip it they will get a warning

logger.warning(
                  "Secret overwrite operation cancelled. Please provide a different name "
                  f"via --{flag}."
)

and this if not secret_name then end the process